management-api-agent-common/src/main/java/io/k8ssandra/metrics/prometheus/CassandraTasksExports.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Please see the included license file for details.
+ */
+package io.k8ssandra.metrics.prometheus;
+
+import com.codahale.metrics.MetricRegistry;
+import com.datastax.mgmtapi.ShimLoader;
+import com.google.common.collect.Lists;
+import io.k8ssandra.metrics.builder.CassandraMetricDefinition;
+import io.k8ssandra.metrics.builder.CassandraMetricNameParser;
+import io.k8ssandra.metrics.config.Configuration;
+import io.prometheus.client.Collector;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.cassandra.db.compaction.OperationType;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Collect non-metrics information from Cassandra and turn them to metrics. This is considerably
+ * slower to collect than the metrics we currently ship.
+ */
+public class CassandraTasksExports extends Collector implements Collector.Describable {
+  private static final org.slf4j.Logger logger =
+      LoggerFactory.getLogger(CassandraTasksExports.class);
+
+  private static final String METRICS_PREFIX = "org_apache_cassandra_metrics_extended_";
+  private final MetricRegistry registry;
+
+  private final CassandraMetricNameParser parser;
+
+  public CassandraTasksExports(MetricRegistry registry, Configuration config) {
+    this.registry = registry;
+    parser = CassandraMetricNameParser.getDefaultParser(config);
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+
+    ArrayList<MetricFamilySamples> familySamples = Lists.newArrayList();
+
+    // Collect Compaction Task metrics
+    familySamples.addAll(getCompactionStats());
+
+    // Collect active streaming sessions
+    familySamples.addAll(getStreamInfoStats());
+
+    // Collect other sstableOperations (if not part of Compactions metrics already)
+
```

Comment:
**Todo:** we need follow up tickets for these missing types of operations

Reply:
At least on 4.1 we have everything, for example UpgradeSSTables is here:

https://github.com/apache/cassandra/blob/8cc2b481922dd77a9956e76aca1220aeb799953b/src/java/org/apache/cassandra/db/compaction/CompactionManager.java#L610